File: examples/delay_except.pl
use v5.14;

use Mojo::IOLoop;

# Handle exceptions in all steps
Mojo::IOLoop::Delay->new->steps(
    sub {
        my $delay = shift;
        die 'Intentional error';
    },
    sub {
        my ($delay, @args) = @_;
        say 'Never actually reached.';
    }
)->catch(sub {
    my ($delay, $err) = @_;
    say "Something went wrong: $err";
})->wait;

File: examples/delay_except.py
import Pyjo.IOLoop

# Handle exceptions in all steps
delay = Pyjo.IOLoop.delay()


@delay.step
def step1(delay):
    raise Exception('Intentional error')


@delay.step
def step2(delay):
    print('Never actually reached.')


@delay.catch
def error(delay, err):
    print("Something went wrong: {0}".format(err))


delay.wait()
